lib/sequel/plugins/validation_helpers.rb

```diff
@@ -295,7 +295,7 @@ def validates_unique(*atts)
               h = ds.joined_dataset? ? qualified_pk_hash : pk_hash
               ds = ds.exclude(h)
             end
-            errors.add(a, message) unless ds.count == 0
+            errors.add(a, message) unless ds.empty?
```

Comment:
Could `EXISTS` be more performant? Or is that what is run as an effect of `.empty?` ?

FWIW - that is what activerecord produces - https://github.com/rails/rails/blob/8eb051a6e681e0509bbe9f352f5ea3461f15ad5c/activerecord/lib/active_record/associations/collection_association.rb#L236